code/controllers/subsystem/objects.dm

```diff
@@ -20,7 +20,11 @@ var/list/processing_objects = list()
 /datum/subsystem/obj/Initialize()
 	for(var/atom/object in world)
 		if(!(flags & ATOM_INITIALIZED))
+			var/time_start = world.timeofday
 			object.initialize()
+			var/time = (world.timeofday - time_start)
+			if(time > 1 SECONDS)
+				log_debug("Slow object initialize. [object] at [object.x],[object.y],[object.z] took [time] seconds to initialize.")
```

Comment:
```suggestion
				log_debug("Slow object initialize. [object.type] at [object.x],[object.y],[object.z] took [time] seconds to initialize.")
```

Reply:
Personally I'd put in both, just in case